### DISCUSSION TITLE
biomedical named entitiy recognition(NER) with "en_core_sci_lg"

### DISCUSSION THREAD

**Discussion by MaazAmjad** [explosion/spaCy#12529] [Help: Coding & Implementations] [answered]
As described in this question([Scispacy for biomedical named entitiy recognition(NER)](https://stackoverflow.com/questions/59265404/scispacy-for-biomedical-named-entitiy-recognitionner/60834482#60834482)), the "ner" component of the model label them as "ENTITY". How can we recognize specifically tailored terms? For example, change `["Banana" --> ENTITY]` to `["Banana" --> "Food"]`. Spacy "EntityLinker"(https://github.com/egerber/spaCy-entity-linker) does not do in this format. What modifications are required?

I only need to use this model since it can extract most of the entities. I only seek help on how can I change the label "ENTITY" to "Food". An example with code would be extremely helpful.

**#Desired output:**

```
nlp = spacy.load("en_core_sci_lg")

doc = nlp("I ate Apple and Banana")

for en in doc.ents:
print(f"{en.text} ---->{en.label_}")
```

Apple ---> Food

Banana ---> Food

I would like to keep the rest entities tagged as "ENTITIES" and change the tags of entities that I would like to do.

**Comment by svlandeg** [comment, marked as answer]
Hi!

`scispacy` and its models are not maintained by us, but as someone pointed out on that SO thread, you can try some of their other models instead which do have more fine-grained labels: https://allenai.github.io/scispacy/.

In general, there's not a single solution to "just" change the labels of "some" entities while keeping the labels of others. You'll have to either [(re)train](https://spacy.io/usage/training) your models, or run some kind of [rule-based matching](https://spacy.io/usage/rule-based-matching) to identify cases where you'd like to change the label span. How you'd do it exactly depends on your use-case & needs though, so it's difficult to advice on that without more details.
